### PR TITLE
Ab/solidstart onboarding

### PR DESCRIPTION
**Pull request by andreiborza**
Adds onboarding for Solid Start projects.
Screenshot 2024-08-26 at 17 24 07@2x

Closes: https://github.com/getsentry/sentry-javascript/issues/12557

### CODE REVIEW

File: static/app/gettingStartedDocs/javascript/solidstart.tsx [hunk 1]
```diff
@@ -0,0 +1,475 @@
...
+        language: 'bash',
+        code: 'npm install --save @sentry/solidstart',
+      },
+      {
+        label: 'yarn',
+        value: 'yarn',
+        language: 'bash',
+        code: 'yarn add @sentry/solidstart',
+      },
+    ],
```
Comment (chargome) [inline static/app/gettingStartedDocs/javascript/solidstart.tsx:182]: pnpm? 🤔

**Review by chargome — APPROVED** [review]
LGTM!

**Review by AbhiPrasad — APPROVED** [review]

File: static/app/gettingStartedDocs/javascript/solidstart.tsx (hunk 1 above)
Comment (andreiborza) [reply static/app/gettingStartedDocs/javascript/solidstart.tsx:188]: Added

File: static/app/gettingStartedDocs/javascript/solidstart.tsx [hunk 2]
```diff
@@ -0,0 +1,481 @@
...
+              code: getSdkRun(),
+            },
+          ],
+        },
+      ],
+    },
+    {
+      title: t('Upload Source Maps'),
+      description: tct(
+        'To upload source maps to Sentry, follow the [link:instructions in our documentation].',
```
Comment (ArthurKnaus) [inline static/app/gettingStartedDocs/javascript/solidstart.tsx:319]: Does it support the sentry wizard?
Then there would be a util that generates this step:
[getUploadSourceMapStep](https://github.com/getsentry/sentry/blob/master/static/app/components/onboarding/gettingStartedDoc/utils/index.tsx)

**Review by ArthurKnaus — COMMENTED** [review]
Since it is a new platform you will want to add it to the `platformCategories` file:
https://github.com/getsentry/sentry/blob/master/static/app/data/platformCategories.tsx
I see `javascript-solid` is missing there as well...

File: static/app/gettingStartedDocs/javascript/solidstart.tsx [hunk 3]
```diff
@@ -0,0 +1,481 @@
...
+      configurations: getInstallConfig(),
+    },
+  ],
+  configure: (params: Params) => [
+    {
+      type: StepType.CONFIGURE,
+      description: tct(
+        "Initialize Sentry as early as possible in your application's lifecycle.",
+        {code: <code />}
+      ),
```
[the remark this answers is not in the document]
Comment (andreiborza) [reply static/app/gettingStartedDocs/javascript/solidstart.tsx:213]: Ah right, but `tct` requires a second param, I'll update to just `t`.

[the remark this answers is not in the document]
Comment (andreiborza) [reply static/app/gettingStartedDocs/javascript/solidstart.tsx]: Fixed

**Comment by andreiborza** [comment]
> Since it is a new platform you will want to add it to the `platformCategories` file: https://github.com/getsentry/sentry/blob/master/static/app/data/platformCategories.tsx I see `javascript-solid` is missing there as well...

Thanks, I added both.

**Review by ArthurKnaus — APPROVED** [review]

**Review by s1gr1d — APPROVED** [review]

**Pull request merged**